jcl/src/java.base/share/classes/java/lang/invoke/MethodHandle.java

```diff
@@ -823,7 +823,11 @@ private static final MethodHandle resolveInvokeDynamic(Class<?> clazz, String na
 		MethodType type = null;
 
 		try {
-			type = MethodType.fromMethodDescriptorString(methodDescriptor, VM.getVMLangAccess().getClassloader(clazz));
+			try { 
+				type = MethodType.fromMethodDescriptorString(methodDescriptor, VM.getVMLangAccess().getClassloader(clazz));
+			} catch (TypeNotPresentException e) {
+				throwNoClassDefFoundError(e);
```

Comment:
One way to make it easier to reason about this code is to write the call as:
``` 
throw throwNoClassDefFoundError(e);
```
so that readers know that an exception is guaranteed to be thrown here and there is no path to continue to the code below.

Reply:
fixed.